scrape/manager_test.go

```diff
@@ -1509,3 +1511,115 @@ scrape_configs:
 		[]string{fmt.Sprintf("http://%s/metrics", otherJobTargetURL)},
 	)
 }
+
+func TestManagerStopAfterScrapeAttempt(t *testing.T) {
```

Comment:
Just a Nit, but would it make sense to split tests for initial scrape offset and scrape on shutdown?

They seem to be two separate features, not sure if it make sense to tie them together in a single test 🤔